src/webots/gui/WbConsole.cpp

```diff
@@ -397,74 +407,106 @@ void WbConsole::handlePossibleAnsiEscapeSequences(const QString &msg, WbLog::Lev
   int i = msg.indexOf("\033[");
   if (i != -1) {  // contains ANSI escape sequences
     QString html;
-    if (i != 0)
-      html = htmlSpan(msg.mid(0, i), level);  // add the text before the ANSI escape sequence
+    if (i != 0)  // escape code is not at the beginning of the string
+      html = htmlSpan(msg.mid(0, i), level);
     while (1) {
       i += 2;  // skip the "\033[" chars
       QString sequence;
-      while (msg[i] != 'm')
-        sequence += msg[i++];
-      i++;
-
+      while (msg[i] != '\0' && msg[i] != 'm') {  // 'm' is the final char when supporting only colors
+        sequence += msg[i];
+        if (msg[i] == 'J' && msg[i - 1] == '2')  // "2J" is not terminated by 'm', but is still valid
+          break;
+        i++;
+      }
```

Comment:
It's not a safe way to parse a ANSI sequence. See my contribution in `WbController.cpp` for a safe way (implemented from the Wikipedia documentation, testing for `character < '\x40'` to find the end of the sequence).